runtime/coverage.go

```diff
@@ -483,3 +484,45 @@ func (r *CoverageReport) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// MarshalLCOV serializes each common.Location/*LocationCoverage
+// key/value pair on the *CoverageReport.Coverage map, to the
+// LCOV format. Currently supports only line coverage, function
+// and branch coverage are not yet available.
+func (r *CoverageReport) MarshalLCOV() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	for location, coverage := range r.Coverage { // nolint:maprange
+		_, err := buf.WriteString(
+			fmt.Sprintf("TN:\nSF:%s\n", location.ID()),
+		)
```

Comment:
Here and below: Directly write to the buffer using `fmt.Fprintf`, which avoids allocating temporary strings:

```suggestion
		_, err := fmt.Fprintf(buf, "TN:\nSF:%s\n", location.ID()),
```